recipes/sdl_mixer/cmake/conanfile.py

```diff
@@ -0,0 +1,255 @@
+        "vorbis": "stb",
+        "wavpack": False,
+        "xmp": False
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os not in ["Linux", "FreeBSD"]:
+            del self.options.tinymidi
+        if not (self.settings.os == "Windows" or is_apple_os(self)):
+            del self.options.nativemidi
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("sdl/2.28.5", transitive_headers=True, transitive_libs=True)
+        if self.options.flac:
+            self.requires("flac/1.4.2")
+        elif self.options.gme:
+            # TODO: not available on CCI
+            self.requires("gme/x.y.z")
+        if self.options.mpg123:
+            self.requires("mpg123/1.31.2")
+        if self.options.minimp3:
+            self.requires("minimp3/cci.20211201")
+        if self.options.vorbis == "stb":
+            self.requires("stb/cci.20230920")
+        elif self.options.vorbis == "vorbisfile":
+            self.requires("vorbis/1.3.7")
+        elif self.options.vorbis == "tremor":
+            # TODO: not available on CCI
+            self.requires("tremor/1.2.1")
+        if self.options.opus:
+            self.requires("opusfile/0.12")
+        if self.options.modplug:
+            self.requires("libmodplug/0.8.9.0")
+        if self.options.xmp == "libxmp":
+            # TODO: not available on CCI
+            self.requires("libxmp/x.y.z")
+        elif self.options.xmp == "libxmp-lite":
+            # TODO: not available on CCI
+            self.requires("libxmp-lite/x.y.z")
+        if self.options.fluidsynth:
+            # TODO: not available on CCI
+            self.requires("fluidsynth/2.2")
+        if self.options.get_safe("tinymidi"):
+            self.requires("tinymidi/cci.20130325")
+        # https://github.com/libsdl-org/SDL_mixer/blob/release-2.6.3/CMakeLists.txt#L148-L162
+        if self.options.vorbis or self.options.flac or self.options.opus:
+            self.requires("ogg/1.3.5")
+        if self.options.wavpack:
+            # TODO: not available on CCI
+            self.requires("wavpack/x.y.z")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.16 <4]")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        rmdir(self, os.path.join(self.source_folder, "external"))
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.cache_variables["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
```

Comment:
```suggestion
```

Seems unnecessary as Conan does this automatically

Reply:
Thanks! The project uses a new enough CMake policy version (3.16) and indeed does not require the use of `cache_variables`.